delira/utils/config.py

```diff
@@ -51,20 +547,219 @@ def nested_get(self, key, *args, **kwargs):
+            dict like object containing values for config, by default None.
+        fixed_model : dict, optional
+            fixed parameters for model, by default None.
+        fixed_training : dict, optional
+            fixed parameters for training, by default None.
+        variable_model : dict, optional
+            variable parameters for model, by default None.
+        variable_training : dict, optional
+            variable parameters for training, by default None.
+        kwargs:
+            additional arguments added to the config
+        """
+        super().__init__(dict_like=dict_like, **kwargs)
+        self._update("fixed_model", self.generate_dict(fixed_model),
+                     overwrite=True)
+        self._update("fixed_training", self.generate_dict(fixed_training),
+                     overwrite=True)
+        self._update("variable_model", self.generate_dict(variable_model),
+                     overwrite=True)
+        self._update(
+            "variable_training",
+            self.generate_dict(variable_training),
+            overwrite=True)
+        self._version = get_versions()["version"]
+
+    @staticmethod
+    def generate_dict(value):
+        """
+        If value is none an emty dict will be created
+
+        Parameters
+        ----------
         value : Any
-            the corresponding value to set the key to.
+            checked value
+
+        Returns
+        -------
+        Any
+            dict if value is none otherwise value is returned
+        """
+        if value is None:
+            return {}
+        else:
+            return value
+
+    @property
+    def params(self):
+        """
+        Returns a :class:`LookupConfig` with all model and training parameters
+
+        Returns
+        -------
+        :class:`LookupConfig`
+            config with model and training parameters
+        """
+        return LookupConfig(fixed_model=self.fixed_model,
+                            fixed_training=self.fixed_training,
+                            variable_model=self.variable_model,
+                            variable_training=self.variable_training)
+
+    @property
+    def variable_params(self):
+        """
+        Returns a :class:`LookupConfig` with all variable parameters
+
+        Returns
+        -------
+        :class:`LookupConfig`
+            config with variable parameters
+        """
+        return LookupConfig(model=self.variable_model,
+                            training=self.variable_training)
+
+    @variable_params.setter
+    def variable_params(self, new_params: dict):
+        """
+        Update variable parameters from dict like object
+        """
+        # create empty dict
+        if "model" not in new_params:
+            new_params["model"] = {}
+
+        # create empty dict
+        if "training" not in new_params:
+            new_params["training"] = {}
+
+        self.variable_model.update(new_params["model"], overwrite=True)
```

Comment:
Should this also be part of the general Config? I thought a clean way to do this, would be to maintain 4 lists of keys and then build a dict of the config entries in the property method? This would keep the objects in the config without the need to copy them, but would require some additional book-keeping inside the `pop` and `setitem` methods

Reply:
I think we could just remove the setter function completely and use the getter to construct the necessary structure of the parameters?

Reply:
I'm afraid that we can't since without a setter, we don't know what arguments are part of the params and to which param-type they belong